torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -1199,14 +1199,14 @@ at::Tensor XLANativeFunctions::empty_symint(
     c10::optional<bool> pin_memory,
     c10::optional<at::MemoryFormat> /* memory_format */) {
   TORCH_LAZY_FN_COUNTER("xla::");
-  auto size = C10_AS_INTARRAYREF_SLOW(sym_size);
+  // auto size = C10_AS_INTARRAYREF_SLOW(sym_size);
   // PT empty*() are optimizations to avoid initializing the data when it is
   // known it will be completely rewritten. But since for us doing a zero*()
   // does not actually end up doing any memory initialization, we use that and
   // avoid going to CPU for it. A common PT pattern is indeed doing empty() plus
   // s_copy_().
-  return bridge::AtenFromXlaTensor(tensor_methods::full(
-      XlaHelpers::I64List(size), 0, GetXlaDeviceOrCurrent(device),
+  return bridge::AtenFromXlaTensor(tensor_methods::empty(
```

Comment:
Instead of this, you should do what we did in `expand_symint`
```
  c10::optional<at::IntArrayRef> size = c10::asIntArrayRefSlowOpt(sym_size);
  if (size.has_value()) {
    return bridge::AtenFromXlaTensor(tensor_methods::expand(
        bridge::GetXlaTensor(self), torch::lazy::ToVector<int64_t>(*size)));
  } else {
    // at least one of the dimension is symbolic, use the sym_int version of the
    // node
    return bridge::AtenFromXlaTensor(
        tensor_methods::expand_symint(bridge::GetXlaTensor(self), sym_size));
  }
```
and implement `tensor_methods::full_symint`

Reply:
Any reason we want to have 2 cases as opposed to having a unified way that can cover 2 cases?

Reply:
we don't need to run all those dynamic shape logic if input is static

Reply:
done